plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,265 @@
+package jenkins
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/bndr/gojenkins"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Jenkins struct {
+	URL    string
+	User   string
+	Passwd string
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+	// HTTP Timeout specified as a string - 3s, 1m, 1h
+	ResponseTimeout internal.Duration
+	Instance        *gojenkins.Jenkins
+
+	LastbuildFilterInterval internal.Duration `toml:"lastbuild_interval"`
+	JobFilterName           []string          `toml:"job_exclude"`
+}
+
+type byBuildNumber []gojenkins.JobBuild
+
+func (a byBuildNumber) Len() int           { return len(a) }
+func (a byBuildNumber) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byBuildNumber) Less(i, j int) bool { return a[i].Number > a[j].Number }
+
+const sampleConfig = `
+  url = "http://my-jenkins-instance:8080"
+  user = "admin"
+  passwd = "admin"
+  ## Set response_timeout
+  response_timeout = "5s"
+
+  ## Optional SSL Config
+  # ssl_ca = /path/to/cafile
+  # ssl_cert = /path/to/certfile
+  # ssl_key = /path/to/keyfile
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Job & build filter
+  lastbuild_interval = "1h"
```

Comment:
I wasn't sure what this did until I read the code, maybe `max_build_age` would be a better name?

Reply:
Ok, I agree, I'm definitely not good at naming stuff